### ISSUE TITLE
Nextcloud 15.0 Released

### ISSUE THREAD

**Issue by supermamie** [YunoHost-Apps/nextcloud_ynh#147]
Same as issue #145 with one more version :slightly_smiling_face: 
I know #146 is ongoing to merge Nextcloud 14.0.4 but a new issue for a new version is now here.

Thanks a lot for all your work.

**Comment by liberodark** [comment]
NC 15 is a good version im go to see that

**Comment by nath5394** [comment]
Hello, I started to update scripts for Nextcloud 15.0.0. 
You can check my fork here : https://github.com/nath5394/nextcloud_ynh/tree/testing
For the moment it works on my test VM. But it seems I have a few things to fix before opening a pull request : 
I have theese errors in Nextcloud : 

> La base de données a quelques index manquant. L'ajout d'index dans de grandes tables peut prendre un certain temps. Elles ne sont donc pas ajoutées automatiquement. En exécutant "occ db:add-missing-indices", ces index manquants pourront être ajoutés manuellement pendant que l'instance continue de tourner. Une fois les index ajoutés, les requêtes sur ces tables sont généralement beaucoup plus rapides.

    Index "owner_index" manquant dans la table "oc_share".
    Index "initiator_index" manquant dans la table "oc_share".
    Index "fs_mtime" manquant dans la table "oc_filecache".

>Certaines colonnes de la base de données n'ont pas été converties en big int. Changer le type de colonne dans de grandes tables peu prendre beaucoup de temps, elles n'ont donc pas été converties automatiquement. En exécutant 'occ db:convert-filecache-bigint' ces changements en suspens peuvent être déclenchés manuellement. Cette opération doit être exécutée pendant que l'instance est hors ligne. Pour plus d'information, consulter la page de la documentation.

    filecache.mtime
    filecache.storage_mtime

And finally I do not know if the patch have to be updated for 15.0.0. I'll try to understand how they work. But if someone has an advice I'll appreciate.

**Comment by nath5394** [comment]
I noticed on my fork, with version 15.0.0 there isn't the Yunohost squared button. I'll try to understand how it was implemented in the previous versions.

**Comment by maniackcrudelis** [comment]
I created a pull request for your work, so we can see easily the diff and have some automatic tests from our CI.

**Comment by maniackcrudelis** [comment]
By the way, I confirm the error with ynhpanel.js, only with the version 15 of Nextcloud (No problem with the current version 13). Without any modifications of the nginx config file (I included the last change from testing branch.)

The error is,
```
Content Security Policy: The page’s settings blocked the loading of a resource at self (“script-src”). Source: call to eval() or related function blocked by CSP.
```
Although, I can't see anything related to eval() in the script, https://github.com/YunoHost/SSOwat/blob/stretch-stable/portal/assets/js/ynhpanel.js

**Comment by JimboJoe** [comment]
It must have something to do with the custom patches we created for the package.
I had to patch Nextcloud source files to disable CSP level 3 to allow the YunoHost tile to show, and that probably must be adapted...
I'll have a look at it and I'll think about a way to share that work as well.

**Comment by nath5394** [comment]
Thanks for the help, I saw your PR https://github.com/YunoHost-Apps/nextcloud_ynh/pull/149 @JimboJoe  so I reverted my last commit wich lauched the same occ commands.

**Comment by liberodark** [comment]
Thanks for NC 15 :D

**Comment by jhammons** [comment]
Thank you all.

**Comment by alexAubin** [comment]
Closing because PR for 15.x has been merged ;)
